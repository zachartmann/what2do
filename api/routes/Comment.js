import { Router } from "express";
import { OK } from "http-status-codes";
import CommentModel from "../models/commentSchema";

const router = Router();

/**
 * GET: /comments
 */

router.get("/comments", async (req, res) => {
  const comments = await CommentModel.find({});

  return res.status(StatusCodes.OK).json(comments);
});

/**
 * POST: /comments
 */

router.post("/comments", async (req, res) => {
  const { commentId, comment, user } = req.query;
  let commentToCreate;

  if (req.query) {
    commentToCreate = new CommentModel({
      commentId,
      comment,
      user,
    });
  } else {
    commentToCreate = new CommentModel({
      commentId: 1,
      comment: "Testing this as a comment this as a comment",
      user: "Testuser",
    });
  }

  commentToCreate.save((err) => {
    if (err) {
      console.log("issue with the comment endpoint");
      return res.status(StausCodes.OK).json(commentToCreate);
    } else {
      console.log("This comment endpoint worked");
    }
  });

  return res
    .status(StatusCodes.INTERNAL_SERVER_ERROR)
    .json("Unable to save comment record");
});

export default router;
