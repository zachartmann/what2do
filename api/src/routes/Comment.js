import { Router } from "express";
import { StausCodes } from "http-status-codes";
import CommentModel from "../models/commentSchema";

const router = Router();

/**
 * GET: /comments
 */

router.get("/comments", async (req, res) => {
  const comments = await CommentModel.find({});

  return res.status(StatusCodes.OK).json(comments);
});

router.get("/comments/:id", async (req, res) => {
  const commentId = Number(req.params.id);
  if (!commentId) {
    return res
      .status(StatusCodes.BAD_REQUEST)
      .json("Comment id is a valid number reference");
  }

  const comments = await CommentModel.findOne({ commentId });

  if (comments) {
    return res.status(StatusCodes.OK).json(comments);
  }

  return res
    .status(StatusCodes.INTERNAL_SERVER_ERROR)
    .json("There is an internal issue with the code");
});

/**
 * POST: /comments
 */

router.post("/comment", async (req, res) => {
  const { commentId, comment, user } = req.query;
  let commentToCreate;

  if (req.query) {
    commentToCreate = new CommentModel({
      commentId,
      comment,
      user,
    });
  } else {
    commentToCreate = new CommentModel({
      commentId: 1,
      comment: "Testing this as a comment this as a comment",
      user: "Testuser",
    });
  }

  commentToCreate.save((err) => {
    if (err) {
      console.log("issue with the comment endpoint");
      return res.status(StausCodes.OK).json(commentToCreate);
    } else {
      console.log("This comment endpoint worked");
    }
  });

  return res
    .status(StatusCodes.INTERNAL_SERVER_ERROR)
    .json("Unable to save comment record");
});

export default router;
