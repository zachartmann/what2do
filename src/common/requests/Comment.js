import axios from "axios";
import { commentsEndpoint, commentEndpoint } from "./Endpoints";

/**
 * Comments
 */

export async function getComments() {
  try {
    const response = await axios.get(commentsEndpoint);
    return response;
  } catch (err) {
    throw err;
  }
}

export async function postComment(comment, user) {
  try {
    const response = await axios.post(commentEndpoint, {
      comment,
      user,
    });
    return response;
  } catch (err) {
    throw err;
  }
}

//commentID was present before maybe should be replaced with _id
//or look at the async function to getIdeas to include the id and ids
