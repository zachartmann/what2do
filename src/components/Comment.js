import React from "react";
//import CommentBox from "../components/CommentBox";


const Comment = ({ comment }) => {

    function buttonPopup() {
        alert('You clicked this button!');
    }

    return (
        <div className="content-container idea">
        <div className="content-container flex-container">
            <div className="flex-component flex-70 flex-container">
            <div className="flex-compoment flex-70">
                <input placeholder="Enter your comment"></input>
            </div>
            </div>
        </div>
        <div className="content-container flex-container">
            <div className="flex-component flex-70 flex-container">
            <div className="flex-compoment flex-70">
                <input placeholder={comment.commentInput}></input>
            </div>
            </div>
        </div>
        <div class="clickbuttonpopuphappens">
                <button onClick={buttonPopup}>ButtonClick</button>
        </div>
        {/*}
        < CommentBox />*/}
        </div>
    )  
}

export default Comment;
