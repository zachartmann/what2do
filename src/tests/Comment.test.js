import { shallow } from "enzyme";
import Comment from "../components/Comment";

const dummyComment = {
  user: "Juanitajhh",
  commentInput: "Just testhing these things",
};

describe("Comment component", () => {
  it("should render correctly with given comment", () => {
    const component = shallow(<Comment key={0} idea={dummyComment} />);

    expect(component).toMatchSnapshot();
  });
});
