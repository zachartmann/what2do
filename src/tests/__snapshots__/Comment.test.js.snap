// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component should render correctly with given comment 1`] = `
<div
  className="content-container comment"
>
  <div
    className="content-container"
  >
    Comment: {commentInput}
    <br />
    User Account: Juanita
  </div>
  <div 
    className="content-container trimmed"
  >
    <button
      className="delete-button"
      onClick={() => deleteComment(commentInput)}
    >
      Delete
    </button>
   </div>
  <br />
</div>
`;
